elastic/kibana
# ## Risk score from new Risk Engine showing in UI

## Risk score from new Risk Engine showing in UI

What happened in this pr:

1. We create the latest transform and index on the `init` call when we install resources for Risk Engine. The original plan was to just get some API layer around our datastream with historical data. But it's not possible in one all to achieve pagination/sorting/filtering of risk scores, so we decided to create transforms.

Latest transform: `risk_score_latest_transform_${spaceId}`
Latest Index: `risk-score.risk-score-latest-${spaceId}`

2. To get the risk score to UI we use the existing search strategy from the old risk score module, and just pass the new index to the search

3. UI are the same except for the single host/user risk score page, when we change the explanation parts and instead of the old UI, we will show alerts table with grouping etc.

<img width="1365" alt="Screenshot 2023-08-09 at 16 19 20" src="https://github.com/elastic/kibana/assets/7609147/0a850b2e-d3d5-4b06-948d-c129dbf754f0">


4. Temporarily pass experimentalFeutres to rule wrapper and bulk create as we need to know, which index to use for alert enrichment on ingest time. It will be removed after we decide to release a new Risk Engine

5. Limiting to have only 2 risk scores per kibana
<img width="972" alt="Screenshot 2023-08-10 at 16 00 42" src="https://github.com/elastic/kibana/assets/7609147/9cc3c545-2ace-42d9-a2f3-ff771c7e5abd">
Because of limited timeframe before FF, majority of UI tests will be added after FF

## How to test

`xpack.securitySolution.enableExperimental: ['riskScoringRoutesEnabled']  `

- Go to Settings -> Entity 

Risk Score
- Enable risk score module
- Generate some alerts with host.name or user.name
- Call from Kibana console calculation API
```
POST kbn:/api/risk_scores/calculation
{
      "data_view_id": ".alerts-security.alerts-default",
      "identifier_type": "user",
      "range": { "start": "now-30d", "end": "now" }
  }
  POST kbn:/api/risk_scores/calculation
{
      "data_view_id": ".alerts-security.alerts-default",
      "identifier_type": "host",
      "range": { "start": "now-30d", "end": "now" }
  }
```
- Go to Security / Explore / Hosts / Hosts Risk and see risk scores
- - If host page not available because it's required integrations, easy fix to create filebeat index
```
PUT filebeat-8.10
{
  "mappings": {
    "properties": {
      "@timestamp": {
        "type":"date"
      },
      "host": {
        "type": "object", 
         "properties": {
           "name": {
             "type": "keyword"
           }
         }
      }
    }
  }
}
```
- Click on any and go to the single host/user risk page and go to Host/User risk tab
- Observe the alerts table for top risk core contributors
